interfaz_controller/interfaz_controller.py

```diff
@@ -1,21 +1,19 @@
 import xlwings as xw
 
 
-def main():
+def main(workbook: xw.Book = None):
     wb = xw.Book.caller()
     sheet = wb.sheets[0]
-    if sheet["A1"].value == "Hello xlwings!":
-        sheet["A1"].value = "Bye xlwings!"
-    else:
-        sheet["A1"].value = "Hello xlwings!"
-
-
-@xw.func
-def hello(name):
-    return f"Hello {name}!"
+    area = float(sheet['B4'].value)
+    h = float(sheet['B5'].value)
+    poro = float(sheet['B6'].value)
+    sw = float(sheet['B7'].value)
+    boi = float(sheet['B8'].value)
```

Comment:
It is better to load all this data at once using a numpy array option or dataframe converter. If you do this, excel calls xlwings for each cell which is inefficient.